CCTPRelayer/src/CCTPRelayer.sol

```diff
@@ -287,6 +287,102 @@ contract CCTPRelayer is ICCTPRelayer, Initializable, UUPSUpgradeable, Ownable2St
         emit PaymentForRelay(nonce, feeAmount);
     }
 
+    function swapAndRequestCCTPWithSolanaSwap(
+        address inputToken,
+        uint256 inputAmount,
+        bytes memory swapCalldata,
+        uint32 destinationDomain,
+        bytes32 mintRecipient,
+        address burnToken,
+        uint256 feeAmount,
+        bytes32 destinationCaller,
+        bytes memory solanaSwapPayload
+    ) external payable nonReentrant {
```

Comment:
The reason this doesn't call `swapAndRequestCCTPTransferWithCaller` and instead has the logic copied and pasted is because it was reverting if i did that so.. this is easier